Importar la  librería ChromaDB

In [1]:
import chromadb

Creamos un chroma_client para que pueda utilizar las funciones y métodos definidos en la clase Client para interactuar con la base de datos ChromaDB.

In [2]:
chroma_client= chromadb.Client()

listar  las colecciones disponibles en la base de datos del cliente, aun no creamos por eso esta vacia

In [3]:
chroma_client.list_collections()

[]

Creamos una nueva colección en la base de datos con el nombre 'prueba2'. 

In [4]:
collection = chroma_client.create_collection(name='prueba2')

Verificamos que se creo la colección

In [5]:
chroma_client.list_collections()

[Collection(id=54343c5d-73ef-44f7-8976-d5050a9f4c22, name=prueba2)]

Cargamos el corpus de los vinos

In [6]:
import pandas as pd

wine_df = pd.read_csv("data/winemag-data-130k-v2.csv")
#wine_df

guardamos en un dataframe los indices y la columna descripcion

In [7]:
corpus = wine_df[['Unnamed: 0','description']]

In [8]:
corpus['description'].shape

(129971,)

Guardamos en una lista la columna descripcion y la columna de índices.

Por ultimo los añadimos a la colección

In [11]:
from tqdm import tqdm
# Supongamos que 'descriptions' ya contiene las descripciones como una lista
descriptions = corpus['description'].tolist()

# Supongamos que 'identificadores' es una lista de identificadores correspondiente
identificadores = corpus['Unnamed: 0'].astype(str).tolist()

for description, identificador in tqdm(zip(descriptions, identificadores), total=len(descriptions)):
    collection.add(
        documents=[description],
        ids=[identificador]
    )    

# Añade los documentos y los ids a la colección
#collection.add(
#    documents=descriptions,
#    ids=identificadores
#)

  0%|          | 437/129971 [00:19<1:34:17, 22.90it/s]


KeyboardInterrupt: 

Mandamos la query a la colección que en este caso es GPT y también el número de resultados con mayor similitud que queremos obtener

In [ ]:
results =  collection.query(
    query_texts=["gpt"],
    n_results=3
)
results

{'ids': [['2', '14', '23']],
 'distances': [[1.7244436740875244, 1.7704598903656006, 1.8153921365737915]],
 'metadatas': [[None, None, None]],
 'embeddings': None,
 'documents': [['Tart and snappy, the flavors of lime flesh and rind dominate. Some green pineapple pokes through, with crisp acidity underscoring the flavors. The wine was all stainless-steel fermented.',
   'Building on 150 years and six generations of winemaking tradition, the winery trends toward a leaner style, with the classic California buttercream aroma cut by tart green apple. In this good everyday sipping wine, flavors that range from pear to barely ripe pineapple prove approachable but not distinctive.',
   'This wine from the Geneseo district offers aromas of sour plums and just enough cigar box to tempt the nose. The flavors are a bit flat at first, then the acidity and tension of sour cherries emerges in the midpalate, bolstered by some black licorice.']],
 'uris': None,
 'data': None,
 'included': ['metadatas'

Los mas cercanos Son los indices 2, 4 y 23. Con las distancias [[1.7244436740875244, 1.7704598903656006, 1.8153921365737915]]

In [ ]:
# Desconectar del servidor
chroma_client.disconnect()